### PR TITLE
Provide raw counters metrics (#5735)

### LINKED ISSUE #5735: Provide raw counters metrics
**Is your feature request related to a problem? Please describe.**
Pulsar is offering many [built-in metrics](https://pulsar.apache.org/docs/en/reference-metrics/), which is pretty awesome. You can find metrics on any layer, from the pulsar-proxy to Bookkeeper. There are easy to plot within Grafana as most of the rate are already computed, for example `pulsar_throughput_in` is directly offering **the total throughput of the namespace coming into this broker in bytes/second.**

However, for some usages, metrics based on rate are not enough. For example, to bill users according to messages produced, you cannot rely on a rate, you need the `raw counter` associated.

**Describe the solution you'd like**
I would like to add a new boolean to the configuration `exposeTopicLevelMetricsAsCounterInPrometheus`, which default is false. By setting it to true, **broker will expose new counter metrics**, which will be the **raw counters** used to compute the rates internally. The new metrics could be suffixed by _counter, for example:

* pulsar_throughput_in_counter
* pulsar_in_counter

and so on. We could also do the same work for any metrics which are a rate, but I think we should start with the `topic metrics`.

**Describe alternatives you've considered**
Using the rate to compute the counter is not offering enough precision.

What do you think about it? We can discuss on the name of the parameter and the naming schemas.

### PR DESCRIPTION
**Pull request by PierreZ**
*(If this PR fixes a github issue, please add `Fixes #<xyz>`.)*

Fixes #5735

### Motivation

Add counter based metrics for topics.

### Modifications

Added counter on AbstractTopics and propagate the result to the PrometheusHandler.

### Verifying this change

Added a test

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - The rest endpoints

### Documentation

  - Does this pull request introduce a new feature? (yes)
  - If yes, how is the feature documented? (yes)

### CODE REVIEW

File: pulsar-broker-common/src/main/java/org/apache/pulsar/broker/ServiceConfiguration.java [hunk 1]
```diff
@@ -1297,6 +1297,11 @@
         doc = "If true, export consumer level metrics otherwise namespace level"
     )
     private boolean exposeConsumerLevelMetricsInPrometheus = false;
+    @FieldContext(
+            category = CATEGORY_METRICS,
+            doc = "If true, export raw counter topic level metrics"
+    )
+    private boolean exposeCounterTopicMetricsInPrometheus = false;
```
Comment (sijie) [inline pulsar-broker-common/src/main/java/org/apache/pulsar/broker/ServiceConfiguration.java]: I don't think we need this flag. 

if `exposeTopicLevelMetricsInPrometheus` is true, we can expose this counter. there should be no penalty to do so.

Comment (codelipenghui) [reply pulsar-broker-common/src/main/java/org/apache/pulsar/broker/ServiceConfiguration.java]: Agree with your point of view

Comment (PierreZ) [reply pulsar-broker-common/src/main/java/org/apache/pulsar/broker/ServiceConfiguration.java]: Right, doing the modifications

**Comment by codelipenghui** [comment]
run java8 tests
run integration tests

**Comment by codelipenghui** [comment]
run cpp tests
run integration tests

**Comment by PierreZ** [comment]
Thanks!

**Pull request merged**